pages/api/auth.ts

```diff
@@ -1,16 +1,42 @@
 import { NextApiRequest, NextApiResponse } from 'next';
+import { createAppExtension } from '@lib/appExtensions';
 import { encodePayload, getBCAuth, setSession } from '../../lib/auth';
 
 export default async function auth(req: NextApiRequest, res: NextApiResponse) {
-    try {
-        // Authenticate the app on install
-        const session = await getBCAuth(req.query);
-        const encodedContext = encodePayload(session); // Signed JWT to validate/ prevent tampering
+  try {
+  // Authenticate the app on install
+  const session = await getBCAuth(req.query);
+  const encodedContext = encodePayload(session); // Signed JWT to validate/ prevent tampering
+  const { access_token: accessToken, context } = session;
 
-        await setSession(session);
-        res.redirect(302, `/?context=${encodedContext}`);
-    } catch (error) {
-        const { message, response } = error;
-        res.status(response?.status || 500).json({ message });
-    }
+  const storeHash = context?.split('/')[1] || '';
+
+  await setSession(session);
+
+  /**
+   * For stores that do not have the app installed yet, create App Extensions when app is
+   * installed.
+   */
+
+  const requestBody = await createAppExtension();
+
+  const response = await(fetch(`https://api.bigcommerce.com/stores/${storeHash}/graphql`, {
```

Comment:
we likely don't want to have the full prod api url like this for two reasons 1) it's not very reusable  and 2) perhaps more importantly, since this is a prod url, we cannot test on any of our dev/ testing environments.

I recommend reviewing what we did for checkout - https://github.com/bigcommerce/sample-app-nodejs/blob/automated-billing-beta/pages/api/checkout/%5Bpid%5D.ts#L14.

Basically, we might want to export this to a variable where we can update the base api location (e.g. `api.bigcommerce.com`) accordingly